pkg/agent/apiserver/handlers/multicast/handler.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multicast
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"antrea.io/antrea/pkg/agent/interfacestore"
+	"antrea.io/antrea/pkg/agent/multicast"
+	"antrea.io/antrea/pkg/agent/querier"
+	"antrea.io/antrea/pkg/antctl/transform/common"
+	"antrea.io/antrea/pkg/features"
+	"antrea.io/antrea/pkg/util/k8s"
+)
+
+type Response struct {
+	PodString string `json:"podString,omitempty"`
+	Inbound   string `json:"inbound,omitempty"`
+	Outbound  string `json:"outbound,omitempty"`
+}
+
+func generateResponse(iface *interfacestore.InterfaceConfig, trafficStats *multicast.PodTrafficStats) Response {
+	return Response{
+		PodString: k8s.NamespacedName(iface.PodNamespace, iface.PodName),
+		Inbound:   strconv.FormatInt(int64(trafficStats.Inbound), 10),
+		Outbound:  strconv.FormatInt(int64(trafficStats.Outbound), 10),
+	}
+}
+
+// HandleFunc returns the function which can handle queries issued by 'antctl get multicaststats' command.
+// It will return pod multicast traffic statistics for the local node.
+func HandleFunc(aq querier.AgentQuerier) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if !features.DefaultFeatureGate.Enabled(features.Multicast) {
+			http.Error(w, "Multicast is not enabled", http.StatusServiceUnavailable)
+			return
+		}
+		stats := aq.GetMulticastInfoQuerier().GetPodStats()
+		responses := make([]Response, 0)
+		for iface, trafficStats := range stats {
+			if trafficStats.Inbound == 0 && trafficStats.Outbound == 0 {
```

Comment:
typically a list request should get all result regardless of the stats is 0 or not, otherwise user may think some Pods are not processed correctly, especially if all Pods' stats are 0, which looks like this API is not working at all.

Reply:
updated